sigs.yaml

```diff
@@ -2009,3 +2009,26 @@ workinggroups:
     contact:
       slack: wg-machine-learning
       mailing_list: https://groups.google.com/forum/#!forum/kubernetes-wg-machine-learning
+  - name: Nodeless
+    dir: wg-nodeless
+    mission_statement: >
+      Explore and ultimately propose a nodeless design that can extend a kubernetes cluster to enable pods running on other service platforms.
```

Comment:
I believe this mission statement assumes an answer, when it should be asking a question. I propose:

```
1. Document "critical user journeys" where interactions with nodes cause end user pain. Document the magnitude of the problem, both in terms of the number of users affected and how severe the problem is. This will help the project prioritize the changes proposed.
2. Propose Kubernetes API changes (if any) that more cleanly separate the roles of infrastructure providers and infrastructure consumers.
3. Propose an overall design (or designs, there may be multiple options) for connecting pods with novel compute systems.
```

Reply:
I think Daniel's in the right direction but maybe simpler is possible?

1) to collect evidence and details of the parts of Kubernetes that unnecessarily expose users to infrastructure concepts with which they would rather not be concerned, and the magnitude of the impact
2) to propose a set of changes to Kubernetes to more cleanly isolate the concept spaces

I stole a bunch of your words, @lavalamp, but I hope it is a little simpler.  The point to me is not "pain" but "disinterest".  Or maybe "undue burden", but I guess that's another way to spell "pain".  I also don't think that enabling novel compute systems are the mission here, even if that is a likely result.

I think we should be thinking of nodeless not as "node-free" but "less node".  We should not presuppose the total elimination of nodes, but it might be safe to presuppose a de-emphasis for some class of users.  This has to start from user issues and should introduce AS LITTLE change as possible to overcome those issues in an architecturally sane way.

Reply:
Or maybe even simpler, presuppose that pain exists:

To propose a set of changes to Kubernetes which de-emphasizes infrastructure concepts, such as nodes, for users that do not need to be aware of them.

Reply:
I think that captures the question of user experience and the desired consumption model - and I think that's ultimately a big part of this exploration. However, I think there's clearly a lot to be explored and untangled WRT the plumbing - in essence that's how we ended up here in the first place. Eg. The way VK was designed _presupposes_ proprietary and opaque APIs that need to be adapted to somehow integrate with K8S as we know it today. I think an important question to be explored is - what would it take to make those proprietary APIs redundant? Is that feasible and/or desirable?

So yes: Define the problem space. Define the desired user experience. Explore and elaborate on the suggested implementations and come up with a proposed optimal outcome.